### DISCUSSION TITLE
using-named-pipes-with-aspnetcore-and-httpclient/

### DISCUSSION THREAD

**Discussion by giscus** [andrewlock/blog-comments#409] [Announcements]
# using-named-pipes-with-aspnetcore-and-httpclient/

In this post I show how to create an ASP.NET Core server that listens using Windows Named Pipes, and how to call the server using an HttpClient

https://andrewlock.net/using-named-pipes-with-aspnetcore-and-httpclient/

**Comment by nomba** [comment]
Andrew, thank you so much for sharing this :thumbsup: Do we have anything similar on non-Windows platform, on Linux?

**Reply by thomaslevesque** [reply]
This should work on Linux as well, as far as I can tell

**Reply by kasajian** [reply]
I don't think there's named pipes there.  Is there.   I think it'd use domain sockets.

**Reply by thomaslevesque** [reply]
Named pipes do exist on Linux (see the `mkfifo` command). But I looked at [the code](https://github.com/dotnet/runtime/blob/8cc7586135856499ea5123bbfb71a5df3fa7f717/src/libraries/System.IO.Pipes/src/System/IO/Pipes/NamedPipeServerStream.Unix.cs#L326), and it looks like `NamedPipeServerStream` uses domain sockets, as you mentioned.

**Reply by MithunChopda** [reply]
As per https://learn.microsoft.com/en-us/dotnet/standard/io/how-to-use-named-pipes-for-network-interprocess-communication ".NET on Linux uses Unix Domain Sockets (UDS) for the implementation of these APIs."
So the same API should work on Linux by leveraging UDS.

**Comment by kasajian** [comment]
Does anyone know if this actually works with Windows auth?  I thought kestrel didn't support Windows Auth.

**Reply by Tyrrx** [reply]
I think so: 
https://apisof.net/catalog/c206bf11e62ddd48cd51d21cccdea950?fx=net462
https://apisof.net/catalog/5a70114aa87f92c7e363c7436c619113?fx=net462

**Comment by softwareliberationarmy** [comment]
Do you know if named pipes are supported in the Windows docker images, too?

**Comment by jacob-l** [comment]
Great article!
How do you think it would work with the Blazor WebAssembly?

**Reply by andrewlock** [reply]
Short answer, this probably won't work with Blazor we assembly or server - you can't access named pipes from the browser AFAIK, and this doesn't work with web sockets!

**Comment by masatma** [comment]
This looks like not possible to extend to support WebSocket over a shared Named Pipe. It may not be able to replace HTTP communication on local network.

**Comment by HellEraser** [comment]
Very nice article, can the client be just a regular .NET service, or app, i.e. no ASP...
I have an ASP.NET worker service app, where I've implemented the server, it uses a controller, but my client is just a regular .NET 8 worker service, where I am doing the GET, but it always times out after the default 100 seconds...

**Reply by Tyrrx** [reply]
I tried it with the example code and it worked.

**Reply by HellEraser** [reply]
I am using a NET8 client, using worker service model, it keeps timing out, tracing indicates it's hitting the ASP endpoint, localhost:5000 - but not the named pipe, times-out...
